### ISSUE TITLE
[A11Y] amp-list, `list` role should be on the container not the element

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#5513]
The `role=list` should be on the `container_` which is the parent of `role=listitem` children. Currently the role is on the element itself and TalkBack is not announcing items as list items. Alternative approach is using `aria-owns` but that's not worth it.

Also we should fix the examples on https://ampbyexample.com/components/amp-list/ they are invalid HTML and there is no need for the wrapper UL and LI children. 

They should just be
```
    <amp-list width=auto
        height=100
        layout=fixed-height
        src="https://ampbyexample.com/json/examples.json">
      <template type="amp-mustache"
          id="amp-template-id">
        <div>
          <a href={{url}}>{{title}}</a>
        </div>
      </template>
    </amp-list>
```

We should add some text to the example that ensures the author we are using proper aria attributes to make these a list.

If we really like to allow folks to use 'li' as children, then `amp-list` needs to become smarter and not create the container until it has a child and can inspect the child to see if it is a `li` or not and based on that either create a `div` or an `ul`

**Comment by lannka** [comment]
hi @kul3r4 do you want to take a look at the ABE issue?

**Comment by aghassemi** [comment]
Closed by https://github.com/ampproject/amphtml/pull/5740

**Comment by lannka** [comment]
@aghassemi to track the ABE bug, you might want to open an issue [here](https://github.com/ampproject/amp-by-example)
